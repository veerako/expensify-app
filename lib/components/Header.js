'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouterDom = require('react-router-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Header = function Header() {
    return _react2.default.createElement(
        'header',
        null,
        _react2.default.createElement(
            'h1',
            null,
            'Expensify'
        ),
        _react2.default.createElement(
            _reactRouterDom.NavLink,
            { to: '/', activeClassName: 'is-active', exact: true },
            ' Dashboard '
        ),
        _react2.default.createElement(
            _reactRouterDom.NavLink,
            { to: '/create', activeClassName: 'is-active' },
            ' Create page '
        ),
        _react2.default.createElement(
            _reactRouterDom.NavLink,
            { to: '/edit', activeClassName: 'is-active' },
            ' Edit page '
        ),
        _react2.default.createElement(
            _reactRouterDom.NavLink,
            { to: '/help', activeClassName: 'is-active' },
            ' Help page '
        )
    );
};
exports.default = Header;