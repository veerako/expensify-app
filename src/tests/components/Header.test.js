// react-test-renderer
//shallow rendering only renders  given component
//full dom rendering renders child components too
//snapshot allows us to keep a track of changes to data over time

import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';


test('Should render header correctly',()=>{

    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
});

 //expect(wrapper.find('h1').text()).toEqual('Expensify');
    //const renderer = new ReactShallowRenderer();
    //renderer.render(<Header/>);
    //expect(renderer.getRenderOutput()).toMatchSnapshot();