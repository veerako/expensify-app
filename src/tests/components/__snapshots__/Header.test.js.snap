// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
     Dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
     Create page 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/edit"
  >
     Edit page 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
     Help page 
  </NavLink>
</header>
`;
